src/topology/is_locally_homeomorph.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import topology.local_homeomorph
+
+/-!
+# Local homeomorphisms
+
+This file defines local homeomorphisms.
+
+## Main definitions
+
+* `is_locally_homeomorph`: A function `f : X → Y` satisfies `is_locally_homeomorph` if for each
+  point `x : X`, the restriction of `f` to some open neighborhood `U` of `x` gives a homeomorphism
+  between `U` and an open subset of `Y`.
+
+  Note that `is_locally_homeomorph` is a global condition. This is in contrast to
+  `local_homeomorph`, which is a homeomorphism between specific open subsets.
+-/
+
+variables {X Y Z : Type*} [topological_space X] [topological_space Y] [topological_space Z]
+  (g : Y → Z) (f : X →  Y)
+
+/-- A function `f : X → Y` satisfies `is_locally_homeomorph` if
+  each `x : x` is contained in the source of some `e : local_homeomorph X Y`. -/
```

Comment:
Please say explicitly that `f = e` or copy the new explanation from the module docstring.